File: app/src/main/java/com/example/degoogle/data/dao/PackagesDao.java
package com.example.degoogle.data.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.example.degoogle.data.entities.InstalledPackages;

import java.util.List;

@Dao
public interface PackagesDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(InstalledPackages installedPackages);
    @Update
    void update(InstalledPackages installedPackages);
    @Delete
    void delete(InstalledPackages installedPackages);
    @Query("SELECT * FROM InstalledPackages")
    LiveData<List<InstalledPackages>> getAllPackages();
}

File: app/src/main/java/com/example/degoogle/data/PackagesDao.kt
package com.example.degoogle.data

import androidx.room.*
import com.example.degoogle.data.entities.InstalledPackages
import kotlinx.coroutines.flow.Flow

@Dao
interface PackagesDao {
@Insert(onConflict = OnConflictStrategy.IGNORE)
suspend fun insert(installedPackages: InstalledPackages)
@Update
suspend fun update(installedPackages: InstalledPackages)
@Delete
suspend fun delete(installedPackages: InstalledPackages)
@Query("SELECT * from InstalledPackages WHERE appPackage = :appPackage")
fun getItems(appPackage:String): Flow<InstalledPackages>
}